src/main/java/org/folio/rmapi/RMAPIService.java

```diff
@@ -231,13 +231,21 @@ private void addRequestHeaders(HttpClientRequest request) {
   }
 
   public CompletableFuture<Titles> retrieveTitles(FilterQuery filterQuery, Sort sort, int page, int count) {
+
```

Comment:
Small remark, I think it's ok to have two different methods like this:
```
  public CompletableFuture<Titles> retrieveTitles(FilterQuery filterQuery, Sort sort, int page, int count) {
    String path = new TitlesFilterableUrlBuilder()
      .filter(filterQuery)
      .sort(sort)
      .page(page)
      .count(count)
      .build();
    return this.getRequest(constructURL(TITLES_PATH + "?" + path), Titles.class);
  }

  public CompletableFuture<Titles> retrieveTitles(Long providerId, Long packageId, FilterQuery filterQuery, Sort sort, int page, int count) {
    String path = new TitlesFilterableUrlBuilder()
      .filter(filterQuery)
      .sort(sort)
      .page(page)
      .count(count)
      .build();

    return this.getRequest(constructURL(
      (VENDORS_PATH + '/' + providerId + '/' + PACKAGES_PATH + '/' + packageId + '/' + TITLES_PATH + "?") + path), Titles.class);
  }
```

That way it won't be necessary to pass nulls, and to check for nulls, also those methods don't really have code duplication because they just pass parameters into TitlesFilterableUrlBuilder and into getRequest

Reply:
Good point -- I originally started off with 2 separate methods and then consolidated. But may make it easier in long run to keep as separate -- will adjust